src/core/ddsi/src/ddsi_eth.c

```diff
@@ -11,16 +11,14 @@
  */
 #include "ddsi_eth.h"
 
-int ddsi_eth_enumerate_interfaces (ddsi_tran_factory_t fact, enum ddsi_transport_selector transport_selector, ddsrt_ifaddrs_t **ifs)
+int ddsi_eth_enumerate_interfaces (ddsi_tran_factory_t fact, ddsrt_ifaddrs_t **ifs)
 {
     int afs[] = { AF_INET, DDSRT_AF_TERM };
 
     (void)fact;
-    (void)transport_selector;
 
 #if DDSRT_HAVE_IPV6
-    if (transport_selector == DDSI_TRANS_TCP6 ||
-        transport_selector == DDSI_TRANS_UDP6)
+    if (fact->m_kind == DDSI_TRANS_TCP6 || fact->m_kind == DDSI_TRANS_UDP6)
```

Comment:
`ddsi_tran_factory.m_kind` is set to one of the `NN_LOCATOR_KIND_...` values, so I think we can test for `m_kind` to be `NN_LOCATOR_KIND_UDPv6` or `NN_LOCATOR_KIND_TCPv6` here?